pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -852,6 +852,9 @@ func (lb *LbaasV1) EnsureLoadBalancer(clusterName string, apiService *api.Servic
 	for _, host := range hosts {
 		addr, err := getAddressByName(lb.compute, host)
 		if err != nil {
+                        if pool != nil {
```

Comment:
If hosts have multiple names, pool.delete will be called repeatedly. But pool just only one.
If pool.delete has be called, we should add check.